scripts/m0

```diff
@@ -41,6 +41,26 @@ M0_SRC_DIR="${M0_SRC_DIR%/*/*}"
 
 SUDO="${SUDO-sudo -E}"  # the absence of ':' is important
 
+#choose the transport 
+_transport_select() {
```

Comment:
Usually, we don't use interactive way to get such selection from user's input.

We can use command line arguments, or environment variables. Please refer to,
```
253 Environment variables:
254     CONFIGURE_OPTS      Extra options for './configure' command.
255     MAKE_OPTS           Extra options for 'make' command.
256 

```

For example, we can use the "NET_XPRT" as the env variable to pass user's selection.
A user can do:
```
NET_XPRT=libfab ./scripts/m0 run-st 
```

or
```
NET_XPRT=lnet ./scripts/m0 run-st 
```

Then in "scripts/m0", we can check this variable to select a network transport.
If this variable is not defined, we can default to lnet or libfab.